src/galaxy/SystemBody.cpp

```diff
@@ -575,3 +577,54 @@ void SystemBody::ClearParentAndChildPointers()
 	m_parent = 0;
 	m_children.clear();
 }
+
+SystemBody *SystemBody::GetNearestJumpable()
+{
+	if (IsJumpable()) return this;
+	// trying to find a jumpable parent
+	SystemBody *result = this;
+	while (result->GetParent()) { // we need to remember the last non-null pointer - this is the root systembody
+		result = result->GetParent();
+		if (result->IsJumpable()) return result;
+	}
+
+	//  Now we climbed to the very top of the hierarchy and did not find a jumpable
+	//  parent - we will have to search purely geometrically
+	//  we go through all the bodies of the system, determining the coordinates
+	//  of jumpable objects, and also remember the coordinates of the current
+	//  object
+	//  the result variable now contains the root systembody
+	std::vector<std::pair<SystemBody *, vector3d>> jumpables;
+	vector3d this_position(0.0);
+	std::function<void(SystemBody *, vector3d)> collect_positions = [&](SystemBody *s, vector3d pos) {
```

Comment:
Instead of using `std::function`, please prefer using `auto collect_positions = []...` - std::function has non-negligible overhead on construction.

Reply:
@Web-eWorks the problem is that when using `auto`, a compilation error occurs as I have a recursive call there

Reply:
Ah, got it. Was expecting recursion to be possible, but thanks C++11!